### PR TITLE
[SPARK-44433][PYTHON][CONNECT][SS][FOLLOWUP]Set back USE_DAEMON after creating streaming python processes

### PR DESCRIPTION
**Pull request by WweiL**
### What changes were proposed in this pull request?

Followup of this comment: https://github.com/apache/spark/pull/42283#discussion_r1283804782 
Change back the spark conf after creating streaming python process.

### Why are the changes needed?

Bug fix

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Config only change

### CODE REVIEW

**Comment by WweiL** [comment]
cc @ueshin Can you take a look? Thank you so much!

This doesn't need to goto 3.5. The change was added with the back port PR: https://github.com/apache/spark/pull/42340/files

File: core/src/main/scala/org/apache/spark/api/python/StreamingPythonRunner.scala [hunk 1]
```diff
@@ -100,7 +106,13 @@ private[spark] class StreamingPythonRunner(
...
   def stop(): Unit = {
     pythonWorker.foreach { worker =>
-      SparkEnv.get.destroyPythonWorker(pythonExec, workerModule, envVars.asScala.toMap, worker)
+      val prevConf = conf.get(PYTHON_USE_DAEMON)
+      conf.set(PYTHON_USE_DAEMON, false)
+      try {
+        SparkEnv.get.destroyPythonWorker(pythonExec, workerModule, envVars.asScala.toMap, worker)
+      } finally {
+        conf.set(PYTHON_USE_DAEMON, prevConf)
+      }
```
Comment (ueshin) [inline core/src/main/scala/org/apache/spark/api/python/StreamingPythonRunner.scala:115]: I don't think this change is needed as `env.destroyPythonWorker` won't refer `conf`.

**Review by ueshin — APPROVED** [review]
LGTM, pending tests.

**Comment by ueshin** [comment]
Thanks! merging to master.

**Pull request closed without merging**